File: listings/child-process/ipc/ipc.py
#!/usr/bin/env python
import socket
import os

print 'NODE_CHANNEL_FD: ', os.environ['NODE_CHANNEL_FD'], '\n'
ipc = socket.fromfd(int(os.environ['NODE_CHANNEL_FD']), socket.AF_UNIX, socket.SOCK_STREAM)

while True:
	line = ipc.recv(256)
	print "child received", line
	ipc.send(line)

File: listings/child-process/ipc/ipc.pl
#!/usr/bin/env perl
use IO::Socket;

print 'NODE_CHANNEL_FD: '.$ENV{'NODE_CHANNEL_FD'};

my $ipc = IO::Socket->new();
$ipc->fdopen($ENV{'NODE_CHANNEL_FD'}, '+<');

while (my $line = $ipc->getline) {
  print "child received: ".$line;
  $ipc->send($line);
}
